### PR TITLE
Index `match` keywords such as `app`.

### LINKED ISSUE #1653: Search for "app match" doesn't result in useful results
I wanted documentation on the the `app` form of `match`. But the search results didn't have anything in it even relating to `match`.  This is likely because `app` is not indexed as it is entirely matched on symbolic name.

https://docs.racket-lang.org/search/index.html?q=app%20match

### PR DESCRIPTION
**Pull request by samth**
Closes #1653.

### CODE REVIEW

**Comment by samth** [comment]
This produces search results that say "app in reference". It's not clear to me whether it's possible to do better than that.

File: pkgs/racket-doc/scribblings/reference/match.scrbl [hunk 1]
```diff
@@ -202,7 +203,7 @@ In more detail, patterns match as follows:
          the patterns @racket[(list-no-order x 1 x)] and
          @racket[(list-no-order x 1 x ...)] both produce syntax errors.}}
 
- @item{@racket[(#,(racketidfont "list-no-order") _pat ... _lvp)] ---
+ @item{@racket[(#,(match-kw "list-no-order") _pat ... _lvp)] ---
```
Comment (sorawee) [inline pkgs/racket-doc/scribblings/reference/match.scrbl:206]: This is the second indexing. Is it OK? (It could be OK, but I'm not sure)

Comment (samth) [reply pkgs/racket-doc/scribblings/reference/match.scrbl:206]: Ah, that was not intended.

Comment (samth) [reply pkgs/racket-doc/scribblings/reference/match.scrbl:206]:
```suggestion
 @item{@racket[(#,(racketidfont "list-no-order") _pat ... _lvp)] ---
```

**Pull request merged**